src/runtime/metatype.cs

```diff
@@ -103,9 +104,16 @@ public static IntPtr tp_new(IntPtr tp, IntPtr args, IntPtr kw)
             var cb = GetManagedObject(base_type) as ClassBase;
             if (cb != null)
             {
-                if (!cb.CanSubclass())
+                try
                 {
-                    return Exceptions.RaiseTypeError("delegates, enums and array types cannot be subclassed");
+                    if (!cb.CanSubclass())
```

Comment:
If this is the only place where `CanSubclass` is used, can it be inlined here, and raising an exception bypassed?

Reply:
Yes, and maybe. `DelegateObject` and `ArrayObject` override the method (always return `false`), we could do a type check here, but I don't think it will improve anything.

Reply:
Just FYI, the reason I am looking for this stuff, is that having exceptions raised and caught interferes with the debugging scenario, when you are debugging a user-visible exception `X`, and to catch its origin set the "Break on all exceptions of type `X`" flag, you will get this line come up.

This particular code is unlikely affect users in this way though.

Reply:
to avoid the exception you could add ClassBase.IsValid and test it explicitly

Reply:
or check cb.type.Valid like you do below

Reply:
Good to have it rewritten with `.Valid`, but not required.